src/Stratis.Bitcoin.Features.Miner/Controllers/MiningController.cs

```diff
@@ -62,16 +51,17 @@ public MiningController(Network network, IPowMining powMining, ILoggerFactory lo
         /// <returns>List of block header hashes of newly mined blocks.</returns>
         /// <remarks>It is possible that less than the required number of blocks will be mined because the generating function only
         /// tries all possible header nonces values.</remarks>
-        [Route("generate")]
+        [Route("startmining")]
```

Comment:
Are changes required for the RPC version of this?

Note that the name change should not apply to the RPC version as there is an expectation for `generate` to exist.

I don't think we currently implement the separate `setgenerate` RPC call. My feeling is that naming this method `StartMining` is somewhat misleading, as it is intended for once-off block generation. The equivalent of `setgenerate`, which would more correctly be called StartMining, is done via a config key that triggers block generation in a loop in the mining feature.

Reply:
Actually good point re generation, let me revert this. We should actually create separate start mining API call? We can have this as a another PR.